s3/libirods_s3.cpp

```diff
@@ -610,16 +630,20 @@ irods::error s3InitPerOperation (
                 rodsLog( LOG_ERROR, "[resource_name=%s] Failed to retrieve S3 region name from resource plugin properties, using 'us-east-1'", resource_name.c_str());
             }
 
+            // TODO: This will never be true as status is not set in this compound conditional
+            // statement whose predicate is status == S3StatusOK.
             if (status != S3StatusOK) {
                 std::string error_str =  boost::str(boost::format("[resource_name=%s] failed to set region name to %s: %s") % resource_name.c_str() %
                     region_name.c_str() % S3_get_status_name(status));
                 rodsLog(LOG_ERROR, error_str.c_str());
                 return ERROR(S3_INIT_ERROR, error_str.c_str());
             }
 
```

Comment:
Agreed on status.  In addition, we are not setting the region name in this routine so the "failed to set region name" error message wouldn't apply anyway.  I think we should torch this part.

The check above is simply to log a warning that we are using the default region.  After that we should just return SUCCESS().

Reply:
Should I change the log message on 630 from LOG_ERROR to LOG_WARNING?

I will remove 633 through 641.

Reply:
Will do